crates/client/src/https_client_connection.rs

```diff
@@ -38,14 +37,29 @@ impl<T> HttpsClientConnection<T> {
     ///
     /// * `name_server` - address of the name server to use for queries
     #[allow(clippy::new_ret_no_self)]
-    pub fn new() -> HttpsClientConnectionBuilder {
-        HttpsClientConnectionBuilder::new()
+    pub fn new() -> HttpsClientConnectionBuilder<T> {
+        HttpsClientConnectionBuilder::default()
+    }
+}
+
+impl<T: TcpConnector> HttpsClientConnection<T> {
+    /// Creates a new client connection with a TCP connector.
+    ///
+    /// *Note* this has side affects of binding the socket to 0.0.0.0 and starting the listening
+    ///        event_loop. Expect this to change in the future.
+    ///
+    /// # Arguments
+    ///
+    /// * `connector` - TCP connector to be used for establishing an HTTPS connection.
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
Since this method is not named `new()`, we probably don't need the clippy attribute?